### PR TITLE
Chore/Dashboard-module

### PR DESCRIPTION
**Pull request by vsantos1711**
Almost selling

### CODE REVIEW

File: back/src/application/dashboard/dashboard.controller.ts
Comment (mathmartins2) [inline back/src/application/dashboard/dashboard.controller.ts:1]: add filter for the dash by condominium id

Comment (mathmartins2) [inline back/src/application/dashboard/dashboard.controller.ts:1]: put await in the promises

File: back/src/application/dashboard/dashboard.service.ts
Comment (mathmartins2) [inline back/src/application/dashboard/dashboard.service.ts:1]: missed types

File: back/src/application/dashboard/interfaces/index.ts
Comment (mathmartins2) [inline back/src/application/dashboard/interfaces/index.ts:1]:
```
import {Resident} from '';

export class ChartDataInterface {
  status?: string;
  month?: string;
  orderCount?: number;
  value?: number;
  receiptDateHour?: Date;
  condominiumName?: string;
  addressee?: Resident;
}

```
i think is best type like this!

File: back/src/application/dashboard/view-model/dashboard-view-model.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        } else {
+          result.push({ condominiumName, value: 1 });
+        }
+      }
+      return result;
+    }, [] as { condominiumName: string; value: number }[]);
+
+    return deliveriesByCondominium;
+  }
+}
```
Comment (mathmartins2) [inline back/src/application/dashboard/view-model/dashboard-view-model.ts]: i think is better move all this logic for the services

File: back/src/application/dashboard/view-model/dashboard-view-model.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      'Abril',
+      'Maio',
+      'Junho',
+      'Julho',
+      'Agosto',
+      'Setembro',
+      'Outubro',
+      'Novembro',
+      'Dezembro',
+    ];
```
Comment (mathmartins2) [inline back/src/application/dashboard/view-model/dashboard-view-model.ts]: i think is better separate this in another file

File: back/src/infra/utils/format-month.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    'Abril',
+    'Maio',
+    'Junho',
+    'Julho',
+    'Agosto',
+    'Setembro',
+    'Outubro',
+    'Novembro',
+    'Dezembro',
+  ];
```
Comment (mathmartins2) [inline back/src/infra/utils/format-month.ts]: monthNames is already defined in `back/src/application/dashboard/view-model/dashboard-view-model.ts`

File: front/src/components/BarChart/function/index.ts
Comment (mathmartins2) [inline front/src/components/BarChart/function/index.ts:1]: missed return type

**Review by mathmartins2 — CHANGES_REQUESTED** [review]

**Comment by vsantos1711** [comment]
Mete bronca

**Review by mathmartins2 — APPROVED** [review]

**Pull request merged**